src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -2886,8 +2889,35 @@ public void setSelectedDownloadableFiles(List<FileMetadata> selectedDownloadable
     public void setSelectedNonDownloadableFiles(List<FileMetadata> selectedNonDownloadableFiles) {
         this.selectedNonDownloadableFiles = selectedNonDownloadableFiles;
     }
-    
-            
+
+    public String getSizeOfDataset() {
+        GetDatasetStorageSizeCommand cmd = new GetDatasetStorageSizeCommand(dvRequestService.getDataverseRequest(), dataset, false, GetDatasetStorageSizeCommand.Mode.DOWNLOAD, workingVersion);
+        try {
+            long bytes = commandEngine.submit(cmd);
+            return FileSizeChecker.bytesToHumanReadable(bytes);
+        } catch (CommandException ex) {
+            return "";
+        }
+    }
+
+    public void validateAllFilesForDownloadArchival() {
+        selectAllFiles();
```

Comment:
I assume this `selectAllFiles()` is what's responsible for the odd UI behavior... when you select "Download" from the Access Dataset dropdown, you get the ZIP download, but now all the files are selected in the files table... that shouldn't be necessary... there has to be a way to get the ZIP without the unexpected UI behavior.

Reply:
I tried adding `clearSelection()` to the end like this...

```
     public void validateAllFilesForDownloadArchival() {
         selectAllFiles();
         boolean guestbookRequired = isDownloadPopupRequired();
         boolean downloadOriginal = false;
         validateFilesForDownload(guestbookRequired, downloadOriginal);
+        clearSelection();
     }
```

... but now it thinks no files are selected:

![Screen Shot 2020-07-02 at 1 08 07 PM](https://user-images.githubusercontent.com/21006/86389919-19df3a00-bc65-11ea-8a3a-2355a96d4b31.png)

So yes, I agree a fix would be nice but I'm not sure what that fix is.

Reply:
Wouldn't the fix be to use a different method than what the one that cares about selected files? i.e

change current method to get selected files and get a list, path that to new private method that takes core of current method but uses this new list.

Create a new method that generates a list and calls new that new private method.

You'd of course have to make sure all the checks for permissions are happening correctly, but I'd expect that to already be in the backend side (i.e. we already shouldn't be relying in the front end to only pass in the permissible values)

Reply:
@scolapasta @mheppler please see 427c88356. This pull request is now much bigger but it no longer uses `selectedFiles` which is what the checkboxes are connected to.